### ISSUE TITLE
Drag and drop added to application (sprint 2)

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/action#297]
## Issue - Enhancement

Let's add Drag-and-Drop to our kanban application. I suggest we use Dan Abramov's [react-dnd](https://github.com/gaearon/react-dnd).
- [x] Added to `/team` dashboard
- [x] Added to `/me` dashboard
- [x] Added to meeting _Project Updates_ section
- [x] Added to agenda queue

Functionality:
- [x] cards can be dragged and reordered within a lane (a column)
- [x] cards can be dragged and reordered between columns
- **Estimated effort:** 5 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by ackernaut** [comment]
I think we can divide (and conquer) via three issues:

- DnD added to Actions List (My Dashboard) —let’s learn!!
- DnD added to Project Columns (as seen on My Dash, Team Dash and Project Updates) —really key!!
- DnD added to Agenda Queue (as seen on Team Dash and Meeting Sidebar) —nice to have!!

**Comment by ackernaut** [comment]
![my-actions-dnd 2x](https://cloud.githubusercontent.com/assets/307286/20107699/13498596-a59f-11e6-86a8-c24c5a44e3a9.png)

**Comment by mattkrick** [comment]
To confirm:
- Cards in the Meeting Agenda Phase cannot be dragged
- Cards in the Team Archive cannot be dragged

@jordanh that's how i'll design it, stop me if i'm wrong

**Comment by ackernaut** [comment]
That sounds right to me!

**Comment by mattkrick** [comment]
TL;DR
- [ ] Can dnd all of the agenda items anywhere, regardless of complete, active, or incomplete status
- [ ] when i'm currently viewing `/5` and someone changes the order to 4, i want to keep the content the same, but change the url.
@jordanh can you confirm that this is the intended behavior? 

Agenda items is going to be tricky. Reason why is because the url (/meeting/agenda/3) is indicative of the sort order. Another fun constraint is that only the incomplete agenda items should be draggable. 
When you combine those, you get a pretty interesting problem. If i have 5 agenda items, I could skip to 3 as the facilitator. Then, I could hit "next" and mark 3 as completed, even though 1 and 2 are incomplete. So this begs the question, why can't i dnd all of them? if i finish 3 before 1, shouldn't i be able to drag 3 to the top? If I can complete items out of order, then i should be able to drag them out of order, too.

Either way, it seems like dividing them in 2 containers (complete, incomplete) and only enabling dnd for the 2nd container is out.

Thankfully, the same method for projects applies here.
So:
- every agenda item is a drag source
- the agenda items div is a drop target
- we can test `canDrop` (if we have to) by looking up the `isComplete` tag. I say let em drag em all!

that's the easy part.
the hard part is making sure the `localPhaseItem` in the url matches the sort order.
To achieve that:
- check if localPhase === AGENDA_ITEMS
- check if nextProps.agenda[localPhaseItem-1].id === oldProps.agenda[localPhaseItem-1].id (in handle redirects)
- if they're different, then find the old one in the nextProps.agenda array & redirect to that localPhase.

doing this means whatever item i'm looking at i'll keep looking at, no matter how wonky your team member sorts stuff.

**Comment by jordanh** [comment]
I finally took the time to read this carefully.

What a PITA!
